components/nexus-core/src/test/java/org/sonatype/nexus/proxy/repository/AnAbstractRepositoryCheckConditionsMethodTest.java

```diff
@@ -158,21 +158,21 @@ public void attributeExternalRequest() throws Exception {
   public void attributeExternalRequestWithFeatureDisabled() throws Exception {
     // HACK: changing final static boolean might not work on all JVMs! It works for me on OSX+Oracle Java7
     // but it might fail as those might be inlined at compile time!
-    Field disabledField = IsItemAttributeMetacontentAttribute.class.getDeclaredField("disabled");
-    disabledField.setAccessible(true);
-    Field modifiers = disabledField.getClass().getDeclaredField("modifiers");
+    Field attributesField = IsRemotelyAccessibleAttribute.class.getDeclaredField("attributes");
+    attributesField.setAccessible(true);
+    Field modifiers = attributesField.getClass().getDeclaredField("modifiers");
     modifiers.setAccessible(true);
-    modifiers.setInt(disabledField, disabledField.getModifiers() & ~Modifier.FINAL);
-    final boolean originalValue = disabledField.getBoolean(null);
+    modifiers.setInt(attributesField, attributesField.getModifiers() & ~Modifier.FINAL);
+    final boolean originalValue = attributesField.getBoolean(null);
     try {
-      disabledField.set(null, true);
+      attributesField.set(null, true);
```

Comment:
However obscure it may seem, we are exposing the ability to disable this feature with a system property. If the property name changes due to refactoring, this test alone does not capture this and protect against a regression.

To protect the property from being susceptible to breakage from internal refactoring or class renaming, suggest writing a test which uses the system property itself. This is a good reason not to use class names when describing a feature that can be controlled by system properties.